coucharchive

```diff
@@ -255,6 +257,35 @@ class ReplicationControl(object):
         self._last_successes = multiprocessing.Manager().list()
         self._last_errors = multiprocessing.Manager().list()
 
+        self.has_run_queue_control = has_run_queue_control
+        if self.has_run_queue_control:
+            def run_queue_control():
+                t = threading.currentThread()
+                while getattr(t, "do_run", True):
+                    time.sleep(2)  # arbitrary
+                    run_queue = max(
+                        _couchdb_request(
+                            source_url + '/_node/_local/_system')['run_queue'],
+                        _couchdb_request(
+                            target_url + '/_node/_local/_system')['run_queue'])
+                    if run_queue > COUCHDB_MAX_RUN_QUEUE:
+                        self.report_error()
+                    logging.debug('CouchDB run_queue: %d' % run_queue)
+
+            self.thread = \
+                threading.Thread(target=run_queue_control)
```

Comment:
This could go in one line :wink: 

Also, maybe `self.run_queue_thread` would be more directly understandable, inside `self.close()`?

Reply:
I changed:
- `self.thread` to `self.run_queue_control_thread`
- `def run_queue_control():` to `self._run_queue_control_function()`

Reply:
:+1: 

(no need for the `_function()` suffix in my opinion, but OK with it too)